models/BaseListing.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Models;
+
+use Components\Connection;
+use DBQueries\SelectQueryBuilder;
+
+/**
+ * Base listing model implementation.
+ */
+final class BaseListing implements Listing
+{
+
+    public function __construct(
+        /**
+         * Database connection.
+         *
+         * @var Connection
+         */
+        private Connection $connection,
+
+        /**
+         * Database table.
+         *
+         * @var string
+         */
+        private string $table,
+
+        /**
+         * Amount of values to get.
+         *
+         * @var int
+         */
+        private int $count = 5,
+
+        /**
+         * Values offset.
+         *
+         * @var int
+         */
+        private int $offset = 0
```

Comment:
It might actually be better to change `Listing` interface so that it does not have to store counts and offsets because it does not seem to be cohesive enough.